catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/impl/OpenSearchFilterVisitor.java

```diff
@@ -380,6 +385,36 @@ public Object visit(Literal expression, Object data) {
     return data;
   }
 
+  protected String normalizePattern(
+      String pattern, String wildcard, String singleChar, String escapeChar) {
+    StringBuilder sb = new StringBuilder(pattern.length());
+    for (int i = 0; i < pattern.length(); i++) {
+      char c = pattern.charAt(i);
+      if (c == escapeChar.charAt(0)) {
+        if (i + 1 < pattern.length()) {
+          i++;
+          String next = Character.toString(pattern.charAt(i));
+          if ("*".equals(next) || "?".equals(next) || "\\".equals(next)) {
```

Comment:
Why not just perform character comparison here and skip the `Character.toString` call?
And how will `next` ever be equal to "\\" if it's only a single character?

Reply:
Good catch. Fixed.